### PR TITLE
AnnotationList: Support html content

### LINKED ISSUE #54423: Annotation-List HTML Content
**Steps to reproduce**:
Create Annotation with a href(a)-Tag and Switch to Annotation List.

**Actual Result**:
The href-Tag is displayed like this: `<a href="http://Server/otherContent">Link-Text</a>` (In raw HTML)

**Expected Result**:
A clickable Link like [Link-Text](http://Server/otherContent).
On the trend panels, hovering over an annotation already displays it correctly.

And same for other HTML-Tag's like `<b>` 

**Environment**:
- Grafana version: v9.1.0-beta1
- User OS & Browser: Win10 & Firefox

### PR DESCRIPTION
**Pull request by lpskdl**
**What this PR does / why we need it**:
Added support for html content inside annotation list item

before change:

https://user-images.githubusercontent.com/108552997/189146042-544383a1-d271-4df3-9da9-7f42a30a672a.mov

after change:

https://user-images.githubusercontent.com/108552997/189146136-a9e4037f-d6f8-472d-9be5-b1a151f522b8.mov

**Which issue(s) this PR fixes**:

Fixes #54423 

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/RenderUserContentAsHTML/RenderUserContentAsHTML.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, { HTMLAttributes, PropsWithChildren } from 'react';
+
+export interface RenderUserContentAsHTMLProps extends Omit<HTMLAttributes<HTMLElement>, 'dangerouslySetInnerHTML'> {
+  component?: React.ElementType;
+  content: string;
+}
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/RenderUserContentAsHTML/RenderUserContentAsHTML.tsx:6]: Tweaked the types here slightly to allow for `<RenderUserContentAsHTML component={Button} content="sample content" />`

Comment (lpskdl) [reply packages/grafana-ui/src/components/RenderUserContentAsHTML/RenderUserContentAsHTML.tsx:6]: Thanks @joshhunt. Can I get a ✅ :)

File: packages/grafana-ui/src/components/RenderUserContentAsHTML/RenderUserContentAsHTML.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export function RenderUserContentAsHTML({
+  component,
+  content,
+  ...rest
+}: PropsWithChildren<RenderUserContentAsHTMLProps>): JSX.Element {
+  return React.createElement(component || 'span', {
+    dangerouslySetInnerHTML: { __html: content },
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/RenderUserContentAsHTML/RenderUserContentAsHTML.tsx:14]: should this be sanitized?

Comment (lpskdl) [reply packages/grafana-ui/src/components/RenderUserContentAsHTML/RenderUserContentAsHTML.tsx]: thank you! Added

**Review by joshhunt — APPROVED** [review]

**Pull request merged**